Source/HYPFormsManager.m

```diff
@@ -556,69 +563,114 @@ - (NSArray *)updateTargets:(NSArray *)targets
 
         if (!field) continue;
 
-        NSArray *fieldIDs = [field.formula hyp_variables];
```

Comment:
Uh, the below part seems like a "major" refactoring which is pretty difficult to be reviewed on a line by line bases. I'm happy to review it for you, but I'd need to have a look at the entire file and compare it to the old version. Would you like me to spend time on this?